airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/job.py

```diff
@@ -339,6 +402,32 @@ def _has_running_concurrent_job(self, errors: Optional[Iterable[Mapping[str, Any
     def _has_reached_max_concurrency(self) -> bool:
         return self._concurrent_attempt == self._concurrent_max_retry
 
+    def _switch_base_url(self) -> None:
+        if self._new_base_url:
+            self.base_url = self._new_base_url
+        else:
```

Comment:
A lot of this flow to get the new shop name, raise `ShopifyBulkExceptions.BulkJobRedirectToOtherShopError`, catch the exception, and then switch the base URL if a new one exists feels pretty complex and confusing.

I'm trying to think if we can streamline this process a bit. I think raising the exception is fair because it is effectively a retry and we need to signal that the stream should perform a retry. Was there a reason why you opted to separate replacing the shop URL into two different methods that need to stay in sync w/ each other as shown by this `else` warning?

If we were to replace the URL in `_should_switch_shop_name()`, then we don't need to check if `_new_base_url` exists and the act of raising the exception to retry would assume that the URL was properly switched.

Reply:
Corrected! Thanks. 

The reason behind this, was to `inform` the customer `from` what we switch `to` what we switch, thus 2 vars were needed. I've managed to have 1 instead. Fixed.